<launch>
  <arg  name="smb_name" 
        default="smb261"
        doc="Name of the SMB in the format smb26x (relevant for calibrations)"/>

  <arg  name="camera_calib_path" 
        default="$(find object_detection)/cfg/$(arg smb_name)_camera_model.yaml"
        doc="Path to the camera calibration file"/>

  <arg  name="input_camera_name" 
        default="/versavis/cam0" 
        doc="Name of the camera, i.e. topic prefix. Relevant for topic subscriptions."/>

  <arg  name="lidar_topic" 
        default='/rslidar_points'
        doc="Topic containing the point cloud from the lidar."/>

  <arg  name="object_detection_classes" 
        default="[0,1,11,25,13,74]"
        doc=  "List of the ids of classes for detection. Based on the COCO dataset, refer here: 
              https://github.com/ethz-asl/darknet_catkin/blob/master/data/coco.names 
              NOTICE, class ids start at 0."/>

  <!--Debayer the image-->
  <node pkg="nodelet" type="nodelet" name="debayer"
        args="standalone image_proc/debayer">
    <remap from="image_raw" to="$(arg input_camera_name)/image_raw" />
    <remap from="image_color" to="$(arg input_camera_name)/image_color" />
    <remap from="image_mono" to="$(arg input_camera_name)/image_mono" />
  </node>

  <!--Object detection node-->
  <node name="object_detector" pkg="object_detection" type="node.py" output="screen">

    <!--Input related-->
    <param name="camera_topic" value='$(arg input_camera_name)/image_color'/>
    <param name="camera_info_topic" value='$(arg input_camera_name)/camera_info'/>
    <param name="lidar_topic" value='$(arg lidar_topic)'/>

    <!--Output related-->
    <rosparam file="$(find object_detection)/cfg/output_params.yaml" />
    
    <!--Camera Lidar synchronization related-->
    <param name="camera_lidar_sync_queue_size" value="10"/>
    <param name="camera_lidar_sync_slop" value="0.1"/>

    <!--Config files-->
    <param name="config_dir" value='$(find object_detection)/cfg/'/>

    <!--Point Projector related-->
    <param name="project_config" value='projector_config_SMB264.yaml'/>

    <!--Object detection related-->
    <param name="model" value='yolov5l6'/>
    <param name="device" value='cpu'/>
    <param name="confident" value='0.4'/>
    <param name="iou" value='0.1'/>
    <rosparam param="classes" subst_value="True">$(arg object_detection_classes)</rosparam> >
    <param name="multiple_instance" value='False'/>

    <!--Object localization related-->
    <param name="model_method" value='hdbscan'/>
    <param name="ground_percentage" value='25'/>
    <param name="bb_contract_percentage" value='10.0'/>
  </node>
</launch>